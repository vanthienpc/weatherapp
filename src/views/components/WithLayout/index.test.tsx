import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import withLayout from '.';

const setup = () => {
  const ComposedComponent = () => <h1>Weather Application</h1>;

  const ComponentWithLayout = withLayout(ComposedComponent);

  const wrapper = shallow(<ComponentWithLayout />);

  return { wrapper };
};

it('render should match the snapshot', () => {
  const { wrapper } = setup();
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('render component with header is weather application', () => {
  const { wrapper } = setup();
  expect(wrapper.html()).toContain('Weather Application');
});
