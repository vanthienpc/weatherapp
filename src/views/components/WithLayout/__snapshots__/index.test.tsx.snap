// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should match the snapshot 1`] = `
<Layout>
  <Memo(GlobalStyleComponent) />
  <Header>
    <h1>
      Weather Application
    </h1>
  </Header>
  <Content>
    <ComposedComponent />
  </Content>
  <Footer>
    @2020 Created by Thien Vo
  </Footer>
</Layout>
`;
